src/pages/SignUp/SignUp.js

```diff
@@ -1,8 +1,136 @@
-import React from "react";
-import "./SignUp.scss";
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
+
+import './SignUp.scss';
 
 function SignUp() {
-  return <div>SignUp</div>;
+  const navigate = useNavigate();
+
+  const [hasEmail, setHasEmail] = useState(false);
+  const [values, setValues] = useState({
+    first_name: '',
+    last_name: '',
+    email: '',
+    password: '',
+    cofirm_password: '',
+  });
+
+  const handleInput = e => {
+    const { name, value } = e.target;
+    setValues({ ...values, [name]: value });
+
+    if (values.email) {
+      setHasEmail(false);
+    }
+  };
+
+  const { first_name, last_name, email, password } = handleInput;
+
+  const emailRegExp =
+    /^[0-9a-zA-Z]([-_\\.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_\\.]?[0-9a-zA-Z])*\.[a-zA-Z]{2,3}$/;
+  const passwordRegExp =
+    /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$ %^&*-]).{8,}$/;
+  const cofirmEmail = values.email && !emailRegExp.test(values.email);
+  const cofirmPassword =
+    values.cofirm_password && values.password !== values.cofirm_password;
+  const cofirmRegPassword =
```

Comment:
변수명을 boolean naming convention에 맞게 is를 붙이는 방식으로 수정해주세요!
```suggestion
  const isEmailValid = values.email && !emailRegExp.test(values.email);
  const isRePasswordValid =
    values.cofirm_password && values.password !== values.cofirm_password;
  const isPasswordValid =
```

Reply:
수정 완료!!